src/views/Delegate.vue

```diff
@@ -55,68 +59,83 @@ const form = ref({
 const { profiles, loadProfiles } = useProfiles();
 
 const networkKey = computed(() => web3.value.network.key);
-const space = computed(() => explore.value.spaces[form.value.id]);
 
-const isValid = computed(() => {
+const { loadOwnedEnsDomains, ownedEnsDomains, validEnsTlds, isValidEnsDomain } =
+  useEns();
+
+const isEnsOwnedByWeb3Account = computed(() =>
+  ownedEnsDomains.value.map(d => d.name).includes(form.value.address)
+);
+
+const validateSpaceInput = computed(() => {
+  if (space.value === null) return t('delegate.noValidSpaceId');
+  return false;
+});
+
+const validateToInput = computed(() => {
+  if (form.value.address === '') return false;
+  const address = form.value.address;
+  if (!isValidEnsDomain(address) && !isAddress(address)) {
+    if (address.includes('.'))
+      return `${t('delegate.noValidEns')} ${t(
+        'setup.supportedEnsTLDs'
+      )}: ${validEnsTlds.join(', ')}`;
+    else return t('delegate.noValidAddress');
+  }
+  if (address.toLowerCase() === web3Account.value.toLowerCase())
+    return t('delegate.delegateToSelf');
+  if (isEnsOwnedByWeb3Account.value) return t('delegate.delegateToSelfAddress');
+  return false;
+});
+
+watch(
+  web3Account,
+  () => {
+    loadOwnedEnsDomains();
+  },
+  { immediate: true }
+);
+
+const isValidForm = computed(() => {
   const address = form.value.address;
   return (
-    auth.isAuthenticated.value &&
-    web3Account.value &&
-    (address.includes('.eth') || isAddress(address)) &&
+    (isValidEnsDomain(address) || isAddress(address)) &&
     address.toLowerCase() !== web3Account.value.toLowerCase() &&
-    (form.value.id === '' || explore.value.spaces[form.value.id])
+    (!specifySpaceChecked.value || space.value?.id === form.value.id) &&
+    !isEnsOwnedByWeb3Account.value
   );
 });
 
-async function load() {
+function clearDelegate(id, delegate) {
+  currentId.value = id;
+  currentDelegate.value = delegate;
+  modalOpen.value = true;
+}
+
+watch([networkKey, web3Account], ([valN, prevN], [valA, prevA]) => {
+  if (valN !== prevN) getDelegationsAndDelegates();
+  if (valA?.toLowerCase() !== prevA) getDelegationsAndDelegates();
```

Comment:
I'd also combine these ifs.

```suggestion
  if (valN !== prevN || valA?.toLowerCase() !== prevA) getDelegationsAndDelegates();
```

I think it's not really possible to trigger both cases at the same time, (at least not with metamask) but technically it would run twice then.